core/trino-main/src/main/java/io/trino/server/protocol/ExecutingStatementResource.java

```diff
@@ -118,14 +118,14 @@ public ExecutingStatementResource(
         queryPurger.scheduleWithFixedDelay(
                 () -> {
                     try {
-                        for (Entry<QueryId, Query> entry : queries.entrySet()) {
+                        for (QueryId queryId : queries.keySet()) {
```

Comment:
Is there some implementation detail about `ConcurrentHashMap` that makes iterating over the key set instead of the entry set more efficient or less lock-hungry?

Reply:
Firstly, we don't need value from `entrySet`, only key. Difference in performance is marginal (around 10% if you don't use value or taking value from map by key with keySet)